### PR TITLE
Adds a document explaining how to contribute to the repository.

### PR DESCRIPTION
**Pull request by thisisdhaas**
Document live at http://orchestra.readthedocs.org/en/contributing_doc/contributing.html

### CODE REVIEW

**Comment by thisisdhaas** [comment]
hm, looks like the github is too smart about simlinks and won't render the top-level ReST at https://github.com/unlimitedlabs/orchestra/blob/contributing_doc/CONTRIBUTING.RST.

Thoughts on how to do it right?

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -1,3 +1,88 @@
-Contributing
-============
+How to contribute to Orchestra
+==============================
 
+So you want to get involved in developing Orchestra. Great! We're excited to
+have your support. This document lays down a few ground rules to help the whole
```
Comment (marcua) [inline docs/source/contributing.rst]: ground rules -> guidelines

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -1,3 +1,88 @@
...
+First, if you find a bug in the code or documentation, check out
+our `open issues <https://github.com/unlimitedlabs/orchestra/issues>`_ and
+`pull requests <https://github.com/unlimitedlabs/orchestra/pulls>`_ to see if
+we're already aware of the problem. Also feel free to reach out to us on
+`gitter <https://gitter.im/unlimitedlabs/orchestra>`_ to answer questions at
+any time.
+
+If you've uncovered something new, please `create an issue
+<https://github.com/unlimitedlabs/orchestra/issues>`_ describing the problem.
+Once you've written code that fixes the issue, `create a pull request
```
Comment (marcua) [inline docs/source/contributing.rst]: Once->If

Comment (marcua) [inline docs/source/contributing.rst]: the issue->an issue

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -1,3 +1,88 @@
...
+* Use your feature branch in the pull request. Any changes that you push to
+  your feature branch will automatically be shown in the pull request.
+
+* Keep your pull requests as small as possible. Large pull requests are hard to
+  review. Try to break up your changes into self-contained and incremental pull
+  requests, if need be, and reference dependent pull requests, e.g. "This pull
+  request builds on request #92. Please review #92 first."
+
+* Include unit tests with your pull request. We love tests and use `CircleCI
+  <https://circleci.com/>`_ to check every pull request and commit.
```
Comment (marcua) [inline docs/source/contributing.rst:47]: want to link them to an example of coverage annotations so they can see how to determine that they've covered their new code? OK if you do it later.

alternatively, tell them about make coverage or something?

File: docs/source/contributing.rst [hunk 4]
```diff
@@ -1,3 +1,88 @@
...
+  Check out our tests in ``orchestra/tests`` to see examples of how to write
+  unit tests. Before submitting a PR, make sure that running ``make test`` from
+  the root directory of the repository succeeds.
+
+* Once you submit a PR, you'll get feedback on your code, sometimes asking for
+  a round or two of changes before your PR is accepted. After each round of
```
Comment (marcua) [inline docs/source/contributing.rst]: "a few rounds of changes"

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -1,3 +1,88 @@
...
+  comments, make changes accordingly, then squash all changes for that round
+  into a single commit with a name like 'changes round 0'.
+
+* After your PR is accepted, your should squash all of your changes into a
```
Comment (marcua) [inline docs/source/contributing.rst]: any quick instructions for this?  something like

```
     git checkout master
     git pull
     git checkout your-branch
     git rebase master
     git log
     <find the commit hash that happened immediately before your first commit (the most recent commit hash that isn't your commit)>
     git reset --soft $THAT_COMMIT_HASH$
     git commit -am "A commit message that summarizes your change in a single line"
     git push -f origin your-branch
```

oh whoops---just saw this below.  should it be here or there?  one argument would be to have it where you have it, but include instructions for both rebasing and squashing...

File: docs/source/contributing.rst [hunk 6]
```diff
@@ -1,3 +1,88 @@
...
+  single commit before merging them into the main codebase.
```
Comment (marcua) [inline docs/source/contributing.rst]: "before we can merge them"

**Comment by marcua** [comment]
Maybe just link to the contributing readthedocs from the file?

File: docs/source/contributing.rst (hunk 1 above)
Comment (thisisdhaas) [inline docs/source/contributing.rst]: changed.

File: docs/source/contributing.rst (hunk 2 above)
Comment (thisisdhaas) [inline docs/source/contributing.rst]: changed.

File: docs/source/contributing.rst (hunk 3 above)
Comment (thisisdhaas) [inline docs/source/contributing.rst:47]: I'm adding some text, focusing on `make test_coverage`. Let me know if you think it needs more.

File: docs/source/contributing.rst (hunk 4 above)
Comment (thisisdhaas) [inline docs/source/contributing.rst]: changed in a single round.

File: docs/source/contributing.rst (hunk 5 above)
Comment (thisisdhaas) [inline docs/source/contributing.rst]: I'll add rebase instructions, but like keeping it below.

**Comment by thisisdhaas** [comment]
I tried that out in `orchestra/CONTRIBUTING.md`: let me know what you think.

**Comment by marcua** [comment]
lgtm---i left one branch about changing the url to master, but otherwise, you're good to merge after fix/squash without another review!

**Pull request merged**